### PR TITLE
fix(OverflowTooltip): paragraph overflow

### PR DESCRIPTION
**Pull request by MEsteves22**
- `HvOverflowTooltip` fixed when `paragraphOverflow` is set to `true`: line clamp was not working and the tooltip was not showing.
- New sample added to the `HvSnackbar` docs to showcase `paragraphOverflow`.
- Since the issue was similar, I took this opportunity to refactor `HvAppSwitcher` to leverage `HvOverflowTooltip`: the tooltip was not showing on the app `title` and code was duplicated unnecessarily. 
   - I don't believe there are breaking changes. The only difference is that the `title` now uses a `div` instead of a `HvTypography` as a `span`

Addresses: https://github.com/lumada-design/hv-uikit-react/issues/3810

### CODE REVIEW

**Comment by plagoa** [comment]
I think the use case that lead to this fix actually required a 3 line clamp. Maybe we can have this be set via property and clamped to a maximum of 3 lines. Maybe default to 2 but also accept 3?

File: packages/core/src/components/AppSwitcher/Action/Action.styles.tsx [hunk 1]
```diff
@@ -41,19 +41,10 @@ export const { staticClasses, useClasses } = createClasses(
...
-
-      textAlign: "left",
-
-      overflow: "hidden",
       whiteSpace: "normal",
-      textOverflow: "ellipsis",
-      display: "-webkit-box",
-      "-webkit-line-clamp": "2",
-      "-webkit-box-orient": "vertical",
-      textWrap: "balance",
```
Comment (zettca) [inline packages/core/src/components/AppSwitcher/Action/Action.styles.tsx:54]: Can't we keep `textWrap: "balance"`?

**Review by zettca — APPROVED** [review]
Nice 👌

File: packages/core/src/components/AppSwitcher/Action/Action.styles.tsx (hunk 1 above)
Comment (MEsteves22) [reply packages/core/src/components/AppSwitcher/Action/Action.styles.tsx:54]: Yup, I'll bring in back

**Comment by MEsteves22** [comment]
> I think the use case that lead to this fix actually required a 3 line clamp. Maybe we can have this be set via property and clamped to a maximum of 3 lines. Maybe default to 2 but also accept 3?

Okay, I was not aware of this need. I'll add a prop to set this!

**Comment by MEsteves22** [comment]
As discussed, the line clamp was updated to [3 lines as defined by DS](https://designsystem.hitachivantara.com/6c705d900/p/7761d5-notification/b/82dd7c). The CSS was overridden in the App Switcher component to only use 2 lines.

File: packages/core/src/components/AppSwitcher/Action/Action.styles.tsx (hunk 1 above)
Comment (MEsteves22) [reply packages/core/src/components/AppSwitcher/Action/Action.styles.tsx:54]: Updated ✅

**Pull request merged**